src/components/general/button/index.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { ReactElement } from 'react';
+import styled from 'styled-components';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { ContextualColors } from '../../../styles';
+import { Icon } from '../../../types/icon';
+import { registerIcons } from '../../../registerIcons';
+registerIcons();
+
+const ButtonContainer = styled.button<{ color: keyof ContextualColors }>`
```

Comment:
I'm not too fond of this name tbh, this isn't a container it just is a button. You could call it StyledButton?